### PR TITLE
[TE] Self-Serve tuning flow 02: refactoring edit alerts + misc

### PR DESCRIPTION
**Pull request by newsummit**
In this PR:
- Removing metric details from alert edit view (since its read-only anyway)
- Adding granularity to general alert settings in `manage/alert`
- Changing performance formula in `alerts/performance`
- Some CSS changes

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js [hunk 1]
```diff
@@ -284,7 +226,7 @@ export default Controller.extend({
    */
   prepareFunctions(configGroup, newId = 0) {
     const newFunctionList = [];
-    const existingFunctionList = configGroup.emailConfig ? configGroup.emailConfig.functionIds : [];
+    const existingFunctionList = configGroup && configGroup.emailConfig ? configGroup.emailConfig.functionIds : [];
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: fyi we can just use the _.has from lodash  https://lodash.com/docs/4.17.4#get

It's behaves similarly to `Ember.getWithDefault` but works with non Ember object

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js [hunk 2]
```diff
@@ -516,21 +453,48 @@ export default Controller.extend({
...
     onSubmit() {
-      const currentId = this.get('alertId');
-      const postFunctionBody = this.get('model.alertData');
-      const newGroupName = this.get('newConfigGroupName');
-      const newEmails = this.get('alertGroupNewRecipient');
-      const oldEmails = this.get('selectedConfigGroupRecipients');
+      const {
+        alertData: postFunctionBody,
+        originalConfigGroup
+      } = this.model;
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: do we need to use `this.get('model')` here? do we expect it to update?

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js [hunk 3]
```diff
@@ -542,11 +506,11 @@ export default Controller.extend({
       });
 
       // Exit quietly (showing warning) in the event of error
-      if (emailError || this.get('isDuplicateEmail')) { return; }
+      if (emailError || isDuplicateEmail) { return; }
 
       // Assign these fresh editable values to the Alert object currently being edited
-      Ember.set(postFunctionBody, 'functionName', this.get('alertFunctionName'));
-      Ember.set(postFunctionBody, 'isActive', this.get('isActive'));
+      Ember.set(postFunctionBody, 'functionName', alertFunctionName);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: Perhaps we should use `Ember.run` here to make sure that the template only rerenders once for both `Ember.set`
https://guides.emberjs.com/v2.11.0/applications/run-loop/#toc_how-do-i-tell-ember-to-start-a-run-loop

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js [hunk 4]
```diff
@@ -224,9 +225,15 @@ export default Ember.Route.extend({
             avgData[key].values = allValues;
           });
 
-          // Make custom calculations
-          avgData['responseRate'] = avgData['responseRate'] ? calculateRate(avgData['totalAlerts'], avgData['totalResponses']) : 'N/A';
-          avgData['precision'] = avgData['precision'] ? calculateRate(avgData['totalAlerts'], avgData['trueAnomalies']) : 'N/A';
+          // Gather totals and make custom calculations
+          let pTrue = getTotalValue('trueAnomalies.tot');
+          let pNew = getTotalValue('newTrend.tot');
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js]: can't we use `const` here if we don't expect the variable to change?

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js [hunk 5]
```diff
@@ -253,22 +260,8 @@ export default Ember.Route.extend({
 
       })
       .catch((err) => {
+        console.log('err : ', err);
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js]: Ew haha!

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss [hunk 6]
```diff
@@ -142,7 +161,7 @@ body {
 .te-label {
   font-weight: normal;
   color:  app-shade(black, 9);
-  margin-bottom: 12px;
+  margin: 15px 0 12px 0;
```
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: BLOCKING COMMENT: this is going to change the styling for a lot of labels in the app. Is this a global change at the art-deco/design level?

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js (hunk 1 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: done - thanks!

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js (hunk 2 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: no - not needed. I'll modify.

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js (hunk 4 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js]: for this instance, we do expect it to change, as its within a `availableGroups.forEach` loop.

File: thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js (hunk 5 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alerts/performance/route.js]: :)

File: thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss (hunk 6 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/styles/shared/_styles.scss]: fixed - wasn't that important.

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js (hunk 3 above)
Comment (newsummit) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/edit/controller.js]: Actually, `postFunctionBody` is not being used by the template anywhere, nor is `alertData`, its source object.

**Comment by justYves** [comment]
LGTM

**Pull request merged**